src/convertToObject.js

```diff
@@ -6,7 +6,32 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const mainObj = {};
+
+  const workArr = sourceString.split(';');
+
+  const cleanArr = workArr
+    .map(function (item) {
+      const result = item.replace(/\n/g, '').trim();
+
+      return result;
+    })
+    .filter((item) => item !== '');
+
+  for (const pair of cleanArr) {
+    const index = pair.indexOf(':');
+
+    if (index > -1) {
+      const key = pair.slice(0, index);
+      const trimKey = key.trim();
+      const value = pair.slice(index + 1);
```

Comment:
Similarly to the `key`, you can trim `value` directly without the need for an intermediate variable: `const value = pair.slice(index + 1).trim();`